src/lib/tasks/run-removal.ts

```diff
@@ -85,22 +85,14 @@ export function runRemoval(
   coordinator.toBeRemovedIndexRange.clear();
   coordinator.toBeRemovedTsRange.clear();
 
-  // Keep track of the last index visited. This is outside of the try block so
-  // that we have access to it in the finally block afterwards.
+  // Define index outside of loop so we can access it in the finally clause.
   let index = lowIndex;
 
   try {
     // Track number of steps to reduce calls to the remaining() callback.
     let step = 1;
 
     for (; index <= highIndex; index++) {
-      // Check to make sure we have made at least one step of progress and that
-      // we haven't run for too long without ceding the thread.
-      if (index > lowIndex && step++ % stepsBetweenChecks === 0 &&
```

Comment:
This time check was moved to the bottom of the loop to make it more clear that at least one pass of the loop will have completed before breaking due to time.